csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -481,7 +481,33 @@ internal void SkipGroup(uint startGroupTag)
         /// </summary>
         public double ReadDouble()
         {
-            return BitConverter.Int64BitsToDouble((long) ReadRawLittleEndian64());
+            if (bufferPos + 8 <= bufferSize)
+            {
+                if (BitConverter.IsLittleEndian)
+                {
+                    var result = BitConverter.ToDouble(buffer, bufferPos);
+                    bufferPos += 8;
+                    return result;
+                }
+                else
+                {
+                    var bytes = new byte[8];

```

Comment:
nit: not that important because we don't really have users on big endian I think, but can we avoid the allocation?
Unfortunately stackalloc is not available because as currently the LangVersion is set to 6.

Reply:
If this dependency is no longer true, then can we start using stackalloc and Span and such niceties?